### PR TITLE
Added Commits view and search for Admin

### PR DESCRIPTION
**Pull request by tanya-saroha**
Added Commits tab in Admin sidebar to view commits of all users. Commits can also be searched based on User, Description and Repository name and between 2 dates. Initially previous day's commits of all users will be seen.

### CODE REVIEW

File: app/controllers/admin/commits_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Admin::CommitsController < ApplicationController
+  
+  def index
+    from = params[:from].presence.try(:to_date).try(:beginning_of_day) || Date.yesterday.beginning_of_day
+    to = params[:to].presence.try(:to_date).try(:end_of_day) || Date.yesterday.end_of_day
+    @commits = Commit.all
+    @commits = @commits.in_range(from, to)
```
Comment (pratik14) [inline app/controllers/admin/commits_controller.rb]: No need of this ```@commits = Commit.all```
you can use 
``` @commits = Commits.in_range(from, to)```

File: app/controllers/admin/commits_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Admin::CommitsController < ApplicationController
+  
+  def index
+    from = params[:from].presence.try(:to_date).try(:beginning_of_day) || Date.yesterday.beginning_of_day
+    to = params[:to].presence.try(:to_date).try(:end_of_day) || Date.yesterday.end_of_day
+    @commits = Commit.all
```
Comment (pratik14) [inline app/controllers/admin/commits_controller.rb]: leave a line between  fetching commits and to/from

File: test/controllers/admin/commits_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  describe 'date range is not provided' do
+    test "should display yesterday's commits" do
+      get :index
+      assert_equal assigns(:commits).count, 5
+    end
+  end
+
+  describe 'date range is provided' do
+    test 'should display commits between two dates: case 1' do
```
Comment (pratik14) [inline test/controllers/admin/commits_controller_test.rb]:
```
describe 'date range is provided' do
  context 'should display commits between two dates' do
    it 'commit present in given time range' do
      ....
   end
  it 'commit not present in given time range' do
    ....
  end
end
end
```

File: test/controllers/admin/commits_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    test 'should display commits between two dates: case 2' do
+      get :index, {from: Date.today, to: Date.today + 1}
+      assert_equal assigns(:commits).count, 0
+    end
+  end
+
+  describe 'query is provided' do
+    test 'should search commits when search query present' do
+      get :index, {query: '1'}
+      assert_equal assigns(:commits).count, 1
```
Comment (pratik14) [inline test/controllers/admin/commits_controller_test.rb]: add test case for 
1. search by repo name
2. search by user name
3. search by description

File: app/controllers/admin/commits_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Admin::CommitsController < ApplicationController
+  
+  def index
+    from = params[:from].presence.try(:to_date).try(:beginning_of_day) || Date.yesterday.beginning_of_day
```
Comment (pratik14) [inline app/controllers/admin/commits_controller.rb]: ```from = (params[:from].presence.try(:to_date) || Date.yesterday).beginning_of_day```

same for "to"

File: app/controllers/admin/commits_controller.rb (hunk 2 above)
Comment (tanya-saroha) [reply app/controllers/admin/commits_controller.rb]: okay sir

**Pull request merged**